src/blog/jspi.md

```diff
@@ -0,0 +1,561 @@
+
+Because the mechanisms used when suspending and resuming WebAssembly
+modules are essentially constant time, we don’t anticipate high costs
+in using JSPI — especially compared to other transformation based
+approaches.
+
+There is a constant amount of work needed to propagate the Promise
+returned by the asynchronous API call to the WebAssembly module
+returning a Promise. Similarly, when a Promise is resolved, the
+WebAssembly application can be resumed immediately with constant-time
+overhead.
+
+However, as with other Promise-style APIs in the browser, any time the
+WebAssembly application suspends it will not be ‘woken up’ again
+except by the browser’s event loop. This requires that the execution
+of the JavaScript code that started the WebAssembly computation itself
+returns to the browser.
+
+### Can I use JSPI to suspend JavaScript programs?
+
+JavaScript already has a well developed mechanism for representing
+asynchronous computations: the Promise and the `async` function
+notation. The JSPI is designed to integrate well with this but not to
+replace it.
+
+### Next steps
+
+The JSPI is currently experimental–one should not use it for Web
+applications that are intended to be deployed in production. However,
+it is a _standard track_ effort; which means that it will eventually
+become a standard and we expect this to become a standard implemented
+across all major browsers.
+
+The rest of this post focuses on how to access the JSPI, how to
+develop code that uses it and some examples to try out.
+
+## How can I use it today?
+
+JSPI is being developed on Intel x64 and on ARM 64 architectures. It
+is available for Linux, macOS, Windows and ChromeOS. To test it
+locally, go to chrome://flags in Chrome, search for “Experimental
+WebAssembly JavaScript Promise Integration (JSPI)” and check the
+box. Relaunch as suggested for it to take effect.
+
+You should use at least version `110.0.5473.0` (MacOS) / `110.0.5469.0`
+(Windows,Android) / `110.0.5478.4` (Linux) or ChromeOS to get the latest
+version of the API. We recommend using the Canary channel to ensure
+that any stability updates are applied. In addition, if you wish to
+use Emscripten to generate WebAssembly (which we recommend), you
+should use a version that is at least `3.1.28`.
+
+It is not yet possible to enable the feature for end users, only to
+test it locally by enabling this flag. Eventually we hope to do an
+Origin Trial to enable this feature for origins that want to opt in.
+
+Once the flag is enabled, you should be able to run scripts that use
+JSPI. Below we show how you can use Emscripten to generate a
+WebAssembly module in C/C++ that uses JSPI. If your application
+involves a different language, not using Emscripten for example, then
+we suggest that you look at how the API works you should look at the
+[proposal](https://github.com/WebAssembly/js-promise-integration/blob/main/proposals/js-promise-integration/Overview.md).
+
+### Limitations
+
```

Comment:
Is it worth mentioning here that debugging with devtools is not currently supported?